cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -156,6 +156,10 @@ func (rm *AzureResourceManager) FindResourceGroupForEnvironment(
 	if err != nil && !errors.As(err, &notFoundError) {
 		return "", fmt.Errorf("getting resource group for environment: %s: %w", envName, err)
 	}
+	// remove any managedBy resource groups
```

Comment:
Could you explain why this is needed?

Reply:
// Several Azure resources can create managed resource groups automatically. Here are a few examples:
	// - Azure Kubernetes Service (AKS)
	// - Azure Data Factory
	// - Azure Machine Learning
	// - Azure Synapse Analytics
	// Managed resource groups are created with the same tag as the environment name, leading azd to think there are
	// multiple resource groups for the environment. We need to filter them out.
	// We do this by checking if the resource group is managed by a resource.